Content.Client/GameObjects/Components/Mobs/Actions/ActionAssignments.cs

```diff
@@ -109,16 +112,28 @@ public ActionAssignments(byte numHotbars, byte numSlots)
                         continue;
                     }
                 }
+
                 assignmentsWithoutItem.Add(assignmentEntry);
             }
+
             // reassign without the item for each assignment we found that no longer has an associated item
             foreach (var (assignment, slots) in assignmentsWithoutItem)
             {
                 foreach (var (hotbar, slot) in slots)
                 {
-                    if (!assignment.TryGetItemActionWithItem(out var actionType, out _)) continue;
-                    AssignSlot(hotbar, slot,
-                        ActionAssignment.For(actionType));
+                    if (!assignment.TryGetItemActionWithItem(out var actionType, out _))
+                    {
+                        continue;
+                    }
+
+                    if (actionMenuLocked)
```

Comment:
this works bit better now, the Lockpad icon on the ActionMenu will enable/disable that black sprite leftover on the action slot when you drop an item. However when you pick up an item, it will always get placed in the first empty slot, which means the black sprite gets taken anyway. To change it, will require probably storing information about what ItemActionType is locked to which slot. Better to leave it for another task.